galsim/phase_screens.py

```diff
@@ -0,0 +1,623 @@
+    A + x(B + x(C + x(D)))
+
+    @param x     Where to evaluate polynomial.
+    @param coef  Polynomial coefficients of increasing powers of x.
+    @returns     Polynomial evaluation.  Will take on the shape of x if x is an ndarray.
+    """
+    result = 0
+    for c in coef[::-1]:
+        result = result*x + c
+    return result
+
+
+def horner2d(x, y, coefs):
+    """Evaluate bivariate polynomial using nested Horner's method.
+
+    @param x      Where to evaluate polynomial (first covariate).  Must be same shape as y.
+    @param y      Where to evaluate polynomial (second covariate).  Must be same shape as x.
+    @param coefs  2D array-like of coefficients in increasing powers of x and y.
+                  The first axis corresponds to increasing the power of y, and the second to
+                  increasing the power of x.
+    @returns      Polynomial evaluation.  Will take on the shape of x and y if these are ndarrays.
+    """
+    result = 0
+    for coef in coefs[::-1]:
+        result = result*x + horner(y, coef)
+    return result
+
+
+class OpticalScreen(object):
+    """
+    Class to describe optical aberrations in terms of Zernike polynomial coefficients.
+
+    Input aberration coefficients are assumed to be supplied in units of wavelength, and correspond
+    to the Zernike polynomials in the Noll convention defined in
+    Noll, J. Opt. Soc. Am. 66, 207-211(1976).  For a brief summary of the polynomials, refer to
+    http://en.wikipedia.org/wiki/Zernike_polynomials#Zernike_polynomials.
+
+    @param tip              Tip aberration in units of reference wavelength.  [default: 0]
+    @param tilt             Tilt aberration in units of reference wavelength.  [default: 0]
+    @param defocus          Defocus in units of reference wavelength. [default: 0]
+    @param astig1           Astigmatism (like e2) in units of reference wavelength.
+                            [default: 0]
+    @param astig2           Astigmatism (like e1) in units of reference wavelength.
+                            [default: 0]
+    @param coma1            Coma along y in units of reference wavelength. [default: 0]
+    @param coma2            Coma along x in units of reference wavelength. [default: 0]
+    @param trefoil1         Trefoil (one of the arrows along y) in units of reference wavelength.
+                            [default: 0]
+    @param trefoil2         Trefoil (one of the arrows along x) in units of reference wavelength.
+                            [default: 0]
+    @param spher            Spherical aberration in units of reference wavelength.
+                            [default: 0]
+    @param aberrations      Optional keyword, to pass in a list, tuple, or NumPy array of
+                            aberrations in units of reference wavelength (ordered according to
+                            the Noll convention), rather than passing in individual values for each
+                            individual aberration.
```

Comment:
Sure.  I added:

```
    @param aberrations      Optional keyword, to pass in a list, tuple, or NumPy array of
                            aberrations in units of reference wavelength (ordered according to
                            the Noll convention), rather than passing in individual values for each
                            individual aberration.  Note that aberrations[1] is piston (and not
                            aberrations[0], which is unused.)  This list can be arbitrarily long to
                            handle Zernike polynomial aberrations of arbitrary order.
```